### PR TITLE
pdms: randomly run pdms in e2e test

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Ref https://github.com/tikv/pd/issues/7519

run e2e in local machine
image

test log in ci
https://ci.pingcap.net/blue/organizations/jenkins/tidb-operator-pull-e2e-kind/detail/tidb-operator-pull-e2e-kind/5325/pipeline/65
image

### Tests

- [x] E2E test 
- [ ] Manual test 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: examples/basic/pd-micro-service-cluster.yaml [hunk 1]
```diff
@@ -17,8 +17,8 @@ spec:
     image: alpine:3.16.0
   pd:
     # TODO: use a stable version
-    baseImage: pingcap/pd
-    version: nightly
+    baseImage: hub-new.pingcap.net/orchestration/pd
+    version: v8.0.0
     maxFailoverCount: 0
```
Comment (HuSharp) [inline examples/basic/pd-micro-service-cluster.yaml:22]: As https://github.com/tikv/pd/pull/7819 changed pdms interface, we need to update it when pd 8.0.0 upgrade

File: pkg/manager/member/pd_member_manager.go [hunk 2]
```diff
@@ -80,8 +80,9 @@ func (m *pdMemberManager) Sync(tc *v1alpha1.TidbCluster) error {
 		return nil
 	}
 
-	if tc.Spec.PD.Mode == "ms" && tc.Spec.PDMS == nil {
-		return fmt.Errorf("tidbcluster: [%s/%s]'s enable micro service but pdMS spec is nil, please check `PDMS`", tc.GetNamespace(), tc.GetName())
+	if tc.Spec.PD.Mode == "ms" && tc.Spec.PDMS == nil ||
+		tc.Spec.PDMS != nil && tc.Spec.PD.Mode != "ms" {
```
Comment (csuzhangxc) [inline pkg/manager/member/pd_member_manager.go:84]: do we need to use `(` `)` to make these conditions more clear?

File: pkg/manager/member/pd_ms_member_manager.go [hunk 3]
```diff
@@ -59,37 +59,40 @@ func NewPDMSMemberManager(dependencies *controller.Dependencies, pdMSScaler Scal
...
+	if tc.Spec.PDMS != nil && tc.Spec.PD == nil {
 		klog.Infof("PD Micro Service is enabled, but PD is not enabled, skip syncing PD Micro Service")
 		return nil
 	}
-	if tc.Spec.PD.Mode != "ms" {
-		// remove all micro service components
-		for _, comp := range tc.Spec.PDMS {
+	// remove all micro service components if PDMS is not enabled
+	// PDMS need to be enabled when PD.Mode is ms && PDMS is not nil
+	if tc.Spec.PDMS == nil || tc.Spec.PD != nil && tc.Spec.PD.Mode != "ms" {
```
Comment (csuzhangxc) [inline pkg/manager/member/pd_ms_member_manager.go:69]: add `(` and `)`?

File: tests/pkg/fixture/fixture.go [hunk 4]
```diff
@@ -190,6 +191,14 @@ func GetTidbCluster(ns, name, version string) *v1alpha1.TidbCluster {
 			},
 		},
 	}
+
+	random := rand.Intn(2)
+	if random != 0 && version == utilimage.PDMSImage {
+		// 50% random in pdms mode
+		tc = AddPDMSForTidbCluster(tc)
+	}
```
Comment (csuzhangxc) [inline tests/pkg/fixture/fixture.go:199]: do we need any message to indicate we're running in MS mode? as this is a random behavior, if no information about this, debug test cases may get harder.

Comment (HuSharp) [reply tests/pkg/fixture/fixture.go]: added.

File: tests/pkg/fixture/fixture.go [hunk 5]
```diff
@@ -190,6 +192,15 @@ func GetTidbCluster(ns, name, version string) *v1alpha1.TidbCluster {
 			},
 		},
 	}
+
+	random := rand.Intn(2)
+	if random != 0 && version == utilimage.PDMSImage {
+		log.Logf("[GetTidbCluster] tidbcluster's pd mode is micro-service in this situation, version: %s", version)
```
Comment (csuzhangxc) [inline tests/pkg/fixture/fixture.go:198]: how about including the ns and name?

Comment (HuSharp) [reply tests/pkg/fixture/fixture.go]: done.

**Pull request merged**